consensus/src/messages/mod.rs

```diff
@@ -23,6 +29,128 @@ use crate::error::SubscribeToAddressesError;
 
 mod handlers;
 
+#[derive(Debug, Serialize, Deserialize)]
+pub enum BlockHeaderMessage {
+    Macro {
+        header: MacroHeader,
+        justification: TendermintProof,
+    },
+    Micro {
+        header: MicroHeader,
+        justification: MicroJustification,
+    },
+}
+
+impl BlockHeaderMessage {
+    pub fn ty(&self) -> BlockType {
+        match self {
+            BlockHeaderMessage::Macro { .. } => BlockType::Macro,
+            BlockHeaderMessage::Micro { .. } => BlockType::Micro,
+        }
+    }
+
+    pub fn body_root(&self) -> &Blake2sHash {
+        match self {
+            BlockHeaderMessage::Macro { header, .. } => &header.body_root,
+            BlockHeaderMessage::Micro { header, .. } => &header.body_root,
+        }
+    }
+
+    pub fn split_block(block: Block) -> (BlockHeaderMessage, BlockBodyMessage) {
```

Comment:
I agree that it would be nicer, but it would introduce a dependency from `Block` to `BlockHeaderMessage` and `BlockBodyMessage` which are defined in the `consensus` crate.